src/accelerate/checkpointing.py

```diff
@@ -128,6 +135,8 @@ def load_accelerator_state(
             The current process index in the Accelerator state
         scaler (`torch.cuda.amp.GradScaler`, *optional*):
             An optional *GradScaler* instance to load
+        optimizer_map_location (`torch.device`, *optional*):
+            What device to load the optimizer state onto. By default uses "cpu".
```

Comment:
Should be None by default then either be cpu or "on_device" (which gets interpreted as accelerator.device after).